python/format/src/datasets_test.py

```diff
@@ -0,0 +1,52 @@
+from etils import epath
+from format.src import datasets
+from format.src import errors
+import pytest
+
+
+# End-to-end tests on real data. The data is in `tests/end-to-end/*.json`.
+@pytest.mark.parametrize(
```

Comment:
Another way to write this test might be to write the output of the validator against the examples to "golden files". The test would then just check whether produced output matches expected output for all examples. 

Pro:
 - context is the expected one.
 - this provides nice examples of what the validator produces, which can then be embedded in the doc.
 - Easier for reviewers to check changes.

Con:
 - we would need a script to update the golden files, and run such a script every time the output format changes. But I think that's doable (I have done something similar in a previous project).

Reply:
Also, it's trivial to add examples / tests.

Reply:
My experience in web development with "snapshot files" is that they are very noisy to update. Particularly, at a time where the format API is likely to change very often, I thought the current solution would be a trade-off between noise/validation.

WDYT of starting with the current solution and update unit tests later if we feel the need?

Reply:
SGTM